src/test/shared/cloudformation/templateRegistry.test.ts

```diff
@@ -0,0 +1,179 @@
+
+                assert.ok(testRegistry.getRegisteredTemplate(filename))
+            })
+
+            it ('returns a template if the registry has multiple registered templates, including the template in question', async () => {
+                const filename = normalizePathIfWindows(path.join(tempFolder, 'template.yaml'))
+                await strToYamlFile(goodYaml1, filename)
+                await testRegistry.addTemplateToTemplateData(pathToUri(filename))
+
+                const filename2 = normalizePathIfWindows(path.join(tempFolder, 'template2.yaml'))
+                await strToYamlFile(goodYaml2, filename2)
+                await testRegistry.addTemplateToTemplateData(pathToUri(filename2))
+
+                assert.ok(testRegistry.getRegisteredTemplate(filename))
+                assert.ok(testRegistry.getRegisteredTemplate(filename))
+            })
+        })
+
+        describe('removeTemplateFromRegistry', async () => {
+            it ('removes an added template', async () => {
+                const filename = normalizePathIfWindows(path.join(tempFolder, 'template.yaml'))
+                await strToYamlFile(goodYaml1, filename)
+                await testRegistry.addTemplateToTemplateData(pathToUri(filename))
+
+                assert.strictEqual(testRegistry.registeredTemplates.size, 1)
+
+                testRegistry.removeTemplateFromRegistry(pathToUri(filename))
+                assert.strictEqual(testRegistry.registeredTemplates.size, 0)
+            })
+
+            it ('does not affect the registry if a nonexistant template is removed', async () => {
+                const filename = normalizePathIfWindows(path.join(tempFolder, 'template.yaml'))
+                await strToYamlFile(goodYaml1, filename)
+                await testRegistry.addTemplateToTemplateData(pathToUri(filename))
+
+                assert.strictEqual(testRegistry.registeredTemplates.size, 1)
+
+                testRegistry.removeTemplateFromRegistry(pathToUri('wrong-template.yaml'))
+                assert.strictEqual(testRegistry.registeredTemplates.size, 1)
+            })
+        })
+    })
+
+    describe('DefaultCloudFormationTemplateRegistryListener', async () => {
+        let sandbox: sinon.SinonSandbox
+        let registry: CloudFormationTemplateRegistry
+        let addTemplateStub: sinon.SinonStub<[vscode.Uri], Promise<void>>
+        let removeTemplateStub: sinon.SinonStub<[vscode.Uri], void>
+        const uri = vscode.Uri.parse('asdf')
+
+        beforeEach(() => {
+            registry = new FakeRegistry()
+            sandbox = sinon.createSandbox()
+            addTemplateStub = sandbox.stub(registry, 'addTemplateToTemplateData')
+            removeTemplateStub = sandbox.stub(registry, 'removeTemplateFromRegistry')
+        })
+
+        afterEach(() => {
+            sandbox.restore()
+        })
+
+        it ('can call a function in an onChange event', async () => {
```

Comment:
these tests aren't really adding value

Reply:
They aren't valuable because the tests are shallow: asserting that a function is called says very little.  40 lines of code to check that `DefaultCloudFormationTemplateRegistryListener` calls a function, doesn't give any confidence that anything actually works.